consensus/ssz_types/src/serde_utils/quoted_u64_fixed_vec.rs

```diff
@@ -0,0 +1,117 @@
+//! Formats `FixedVector<u64,N>` using quotes.
+//!
+//! E.g., `FixedVector::from(vec![0, 1, 2])` serializes as `["0", "1", "2"]`.
+//!
+//! Quotes can be optional during decoding. If `N` does not equal the length deserialization will fail.
+
+use crate::FixedVector;
+use serde::ser::SerializeSeq;
+use serde::{Deserializer, Serializer};
+use serde_utils::quoted_u64_vec::QuotedIntWrapper;
+use std::marker::PhantomData;
+use typenum::Unsigned;
+
+pub struct QuotedIntFixedVecVisitor<N> {
+    _phantom: PhantomData<N>,
+}
+
+impl<'a, N> serde::de::Visitor<'a> for QuotedIntFixedVecVisitor<N>
+where
+    N: Unsigned,
+{
+    type Value = FixedVector<u64, N>;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(formatter, "a list of quoted or unquoted integers")
+    }
+
+    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+    where
+        A: serde::de::SeqAccess<'a>,
+    {
+        let mut vec = vec![];
+
+        while let Some(val) = seq.next_element()? {
```

Comment:
I wasn't clear on this before, but I think we should add a safe-guard against growing `vec` greater than `N::to_usize()`. I was tempted to just leave it as is, but some of the vulnerabilities observed in `ssz` have made me weary of over-allocating.

E.g., (I'm not sure if this compiles)

```rust
/// Returns a `Vec` of no more than `max_items` length.
fn deser_max(mut seq: A, max_items: usize) -> Result<Vec<QuotedIntWrapper>, A::Error>
where
        A: serde::de::SeqAccess<'a>,
{
  let mut vec = vec![];
  let mut counter = 0;
  
  while let Some(val) = seq.next_element()? {
    counter += 1;
    if counter > max_items {
      return Err(Something)
    }

    vec.push(val.int);
  }

  Ok(vec)
}
```

Reply:
Yeah, makes sense. Updated 👍